src/signTxPoolRegistration.c

```diff
@@ -60,20 +65,39 @@ static inline void advanceState()
 
 	switch (subctx->state) {
 
-	case STAKE_POOL_REGISTRATION_PARAMS:
-		subctx->state = STAKE_POOL_REGISTRATION_OWNERS;
-		ASSERT(subctx->numOwners >= 1);
+	case STAKE_POOL_REGISTRATION_INIT:
+		subctx->state = STAKE_POOL_REGISTRATION_POOL_KEY;
+		break;
 
-		txHashBuilder_addPoolRegistrationCertificate_enterOwners(txHashBuilder);
+	case STAKE_POOL_REGISTRATION_POOL_KEY:
+		subctx->state = STAKE_POOL_REGISTRATION_VRF_KEY;
+		break;
+
+	case STAKE_POOL_REGISTRATION_VRF_KEY:
+		subctx->state = STAKE_POOL_REGISTRATION_FINANCIALS;
+		break;
+
+	case STAKE_POOL_REGISTRATION_FINANCIALS:
+		subctx->state = STAKE_POOL_REGISTRATION_REWARD_ACCOUNT;
 		break;
 
+	case STAKE_POOL_REGISTRATION_REWARD_ACCOUNT:
+		txHashBuilder_addPoolRegistrationCertificate_enterOwners(txHashBuilder);
+		subctx->state = STAKE_POOL_REGISTRATION_OWNERS;
+
+		if (subctx->numOwners > 0) {
+			break;
+		}
+
+	// intentional fallthrough
+
 	case STAKE_POOL_REGISTRATION_OWNERS:
 		ASSERT(subctx->currentOwner == subctx->numOwners);
-		ASSERT(subctx->numOwnersGivenByPath == 1);
 
+		txHashBuilder_addPoolRegistrationCertificate_enterRelays(txHashBuilder);
 		subctx->state = STAKE_POOL_REGISTRATION_RELAYS;
+
 		if (subctx->numRelays > 0) {
```

Comment:
We should probably display to the operator that the pool has no relays, especially if we hide info about relays from owners, so it's easy to get confused